modules/core/arrow-test/src/main/kotlin/arrow/test/laws/FoldableLaws.kt

```diff
@@ -17,41 +19,47 @@ import io.kotlintest.properties.forAll
 
 object FoldableLaws {
 
-  fun <F> laws(FF: Foldable<F>, cf: (Int) -> Kind<F, Int>, EQ: Eq<Int>): List<Law> =
-    listOf(
-      Law("Foldable Laws: Left fold consistent with foldMap") { FF.leftFoldConsistentWithFoldMap(cf, EQ) },
-      Law("Foldable Laws: Right fold consistent with foldMap") { FF.rightFoldConsistentWithFoldMap(cf, EQ) },
-      Law("Foldable Laws: Exists is consistent with find") { FF.existsConsistentWithFind(cf) },
-      Law("Foldable Laws: Exists is lazy") { FF.existsIsLazy(cf, EQ) },
-      Law("Foldable Laws: ForAll is lazy") { FF.forAllIsLazy(cf, EQ) },
-      Law("Foldable Laws: ForAll consistent with exists") { FF.forallConsistentWithExists(cf) },
-      Law("Foldable Laws: ForAll returns true if isEmpty") { FF.forallReturnsTrueIfEmpty(cf) },
-      Law("Foldable Laws: FirstOption returns None if isEmpty") { FF.firstOptionReturnsNoneIfEmpty(cf) },
-      Law("Foldable Laws: FirstOption returns None if predicate fails") { FF.firstOptionReturnsNoneIfPredicateFails(cf) },
-      Law("Foldable Laws: FoldM for Id is equivalent to fold left") { FF.foldMIdIsFoldL(cf, EQ) }
+  fun <F> laws(FF: Foldable<F>, GENK: GenK<F>): List<Law> =
+    laws(FF, GENK.genK(Gen.intSmall()))
+
+  fun <F> laws(FF: Foldable<F>, GEN: Gen<Kind<F, Int>>): List<Law> {
```

Comment:
Just remove this in favor of the other one. Ideally `GenK` is all we need.

Reply:
You could also keep this and make it internal. These laws are an api and the fact that we are using `Int` is an implementation detail that should not leak out imo

Reply:
true, i have marked them `@deprecated` for now and will remove them as part of finalizing this PR